internal/pkg/agent/application/actions/handlers/handler_action_upgrade.go

```diff
@@ -91,17 +92,31 @@ func (h *Upgrade) Handle(ctx context.Context, a fleetapi.Action, ack acker.Acker
 // ackActions Acks all the actions in bkgActions, and deletes entries from bkgActions.
 // User is responsible for obtaining and releasing bkgMutex lock
 func (h *Upgrade) ackActions(ctx context.Context, ack acker.Acker) {
+	h.ackLock.Lock()
+	defer h.ackLock.Unlock()
```

Comment:
based on the godoc of `ackAction` I would expect here `ackLock` to be `bkgMutex`. Is `ackLock` intentional here, and if yes, could you please help me understand what concurrent pattern we are trying to prevent here? 🙂

Reply:
I think we can probably get rid of the need for the second lock: https://github.com/elastic/elastic-agent/pull/7635/files#r2027652932